legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-physical-plan/src/main/java/org/finos/legend/engine/persistence/components/schemaevolution/SchemaEvolution.java

```diff
@@ -105,6 +105,33 @@ public SchemaEvolution(Sink sink, IngestMode ingestMode, Set<SchemaEvolutionCapa
         this.ingestMode = ingestMode;
         this.schemaEvolutionCapabilitySet = schemaEvolutionCapabilitySet;
         this.ignoreCase = ignoreCase;
+        this.validateCapabilities();
+    }
+
+    private void validateCapabilities()
+    {
+        if (schemaEvolutionCapabilitySet.contains(SchemaEvolutionCapability.DATA_TYPE_LENGTH_CHANGE) && schemaEvolutionCapabilitySet.contains(SchemaEvolutionCapability.DATA_TYPE_LENGTH_CHANGE_ALLOW_INCREMENT_ONLY))
+        {
+            throw new IllegalArgumentException("Invalid schema evolution capabilities. Select either DATA_TYPE_LENGTH_CHANGE or DATA_TYPE_LENGTH_CHANGE_ALLOW_INCREMENT_ONLY.");
+        }
+
+        if (schemaEvolutionCapabilitySet.contains(SchemaEvolutionCapability.DATA_TYPE_SCALE_CHANGE) && schemaEvolutionCapabilitySet.contains(SchemaEvolutionCapability.DATA_TYPE_SCALE_CHANGE_ALLOW_INCREMENT_ONLY))
+        {
+            throw new IllegalArgumentException("Invalid schema evolution capabilities. Select either DATA_TYPE_SCALE_CHANGE or DATA_TYPE_SCALE_CHANGE_ALLOW_INCREMENT_ONLY.");
+        }
+    }
+
+    public boolean isSchemaEvolvable(Dataset mainDataset, SchemaDefinition stagingSchema)
+    {
+        try
+        {
+            buildLogicalPlanForSchemaEvolution(mainDataset, stagingSchema);
+            return true;
+        }
+        catch (IncompatibleSchemaChangeException e)
+        {
+            return false;
```

Comment:
Shall we just throw IncompatibleSchemaChangeException back so that the caller can know better on what happened ?